lib/remap-lang/src/expression/if_statement.rs

```diff
@@ -0,0 +1,43 @@
+use super::Error as E;
+use crate::{value, Expr, Expression, Object, Result, State, Value};
+
+#[derive(thiserror::Error, Debug, PartialEq)]
+pub enum Error {
+    #[error("invalid value kind")]
+    Value(#[from] value::Error),
+}
+
+#[derive(Debug)]
+pub(crate) struct IfStatement {
+    conditional: Box<Expr>,
+    true_expression: Box<Expr>,
+    false_expression: Box<Expr>,
+}
+
+impl IfStatement {
+    pub fn new(
+        conditional: Box<Expr>,
+        true_expression: Box<Expr>,
+        false_expression: Box<Expr>,
+    ) -> Self {
+        Self {
+            conditional,
+            true_expression,
+            false_expression,
+        }
+    }
+}
+
+impl Expression for IfStatement {
+    fn execute(&self, state: &mut State, object: &mut dyn Object) -> Result<Option<Value>> {
+        match self.conditional.execute(state, object)? {
+            Some(Value::Boolean(true)) => self.true_expression.execute(state, object),
+            Some(Value::Boolean(false)) | None => self.false_expression.execute(state, object),
```

Comment:
I'm a little nervous about conflating `None` (undefined) with `false`, but we'll have to see how this plays out. It would probably be worth introducing a `defined` or `undef` function or equivalent later to test specifically for that case.

Reply:
Yeah, we definitely need to play with this to see what works best.